exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/UnlimitedRawBatchBuffer.java

```diff
@@ -32,12 +33,24 @@
   private final int softlimit;
   private final int startlimit;
 
-  public UnlimitedRawBatchBuffer(FragmentContext context, int fragmentCount) {
-    super(context, fragmentCount);
+  private int runtimeSoftLimit = -1;
+  private int runtimeAckCredit = 1;
+  private int sampleTimes = 0;
+  private long totalBatchSize = 0l;
+  private final int fragmentCount;
+  private final int maxSampleTimes;
+  private final long thresholdNetworkMem;
+
+  public UnlimitedRawBatchBuffer(FragmentContext context, int fragmentCount, boolean enableDynamicFC) {
+    super(context, fragmentCount, enableDynamicFC);
     this.softlimit = bufferSizePerSocket * fragmentCount;
     this.startlimit = Math.max(softlimit/2, 1);
     logger.trace("softLimit: {}, startLimit: {}", softlimit, startlimit);
     this.bufferQueue = new UnlimitedBufferQueue();
+    this.fragmentCount = fragmentCount;
+    this.sampleTimes = fragmentCount;
+    this.maxSampleTimes = fragmentCount;
+    this.thresholdNetworkMem = context.getConfig().getLong(ExecConstants.UNLIMITED__BUFFER_MAX_MEMORY_SIZE);
```

Comment:
I'm concerned that this feature is always enabled with a default size of 500MB. If we assume that most batches are usually < 10MB, we've gone from a potential buffer size of 30MB to one of 500MB. In large queries, this could cause unexpected OOM errors.

Maybe define this so that 0 is the default, and 0 means "don't dynamically change buffer size." This allows us to try out the feature, and for you to use the feature, without running the risk of destabilizing other production sites.

Reply:
Since we already have a configure item to control this feature. Why we still need to set the limited buffer size to 0?

Reply:
Sorry, I missed that. There is no documentation here and the path to set the `enableDynamicFc` field is not entirely obvious to this reviewer. Still, there is a config option. so good enough.